retina-agent panics while initializing on Flatcar Alpha with Linux 6.12

**Describe the bug**
I observed the following panic while trying to deploy retina with Flatcar Alpha:
```console
Starting Retina Agent
starting Retina daemon with legacy control plane v1.0.0-rc3
init client-go
2025/10/14 08:47:54 telemetryInterval is not set, defaulting to 15m0s
init logger
ts=2025-10-14T08:47:54.642Z level=info caller=metrics/metrics.go:198 msg="Metrics initialized"
ts=2025-10-14T08:47:54.642Z level=info caller=standard/daemon.go:142 msg="{data aggregation level 15 0 high <nil>}"
ts=2025-10-14T08:47:54.642Z level=info caller=standard/daemon.go:149 msg="telemetry enabled{applicationInsightsID 15 0 99c62452-029a-4341-bc0f-e689b7473601 <nil>}"
ts=2025-10-14T08:47:54.664Z level=info caller=standard/daemon.go:231 msg="Kubernetes server version: v1.32.7"
ts=2025-10-14T08:47:54.664Z level=info caller=pluginmanager/pluginmanager.go:58 msg="plugin manager has pod level disabled"
ts=2025-10-14T08:47:54.665Z level=info caller=controllermanager/controllermanager.go:72 msg="Initializing controller manager ..."
ts=2025-10-14T08:47:54.665Z level=info caller=servermanager/servermanager.go:33 msg="Initializing HTTP server ..."
ts=2025-10-14T08:47:54.665Z level=info caller=server/server.go:42 msg="Setting up handlers"
ts=2025-10-14T08:47:54.665Z level=info caller=server/server.go:57 msg="Completed handler setup"
ts=2025-10-14T08:47:54.665Z level=info caller=servermanager/servermanager.go:37 msg="HTTP server initialized..."
ts=2025-10-14T08:47:54.665Z level=info caller=standard/daemon.go:314 msg="Started controller manager"
ts=2025-10-14T08:47:54.665Z level=info caller=server/server.go:208 msg="Starting metrics server"
ts=2025-10-14T08:47:54.665Z level=info caller=servermanager/servermanager.go:42 msg="Starting HTTP server ..." host=0.0.0.0 port=10093
ts=2025-10-14T08:47:54.665Z level=info caller=server/server.go:247 msg="Serving metrics server" bindAddress=:18080 secure=false
ts=2025-10-14T08:47:54.665Z level=info caller=pluginmanager/pluginmanager.go:118 msg="Starting plugin manager ..."
ts=2025-10-14T08:47:54.666Z level=info caller=pluginmanager/pluginmanager.go:150 msg="Skipping Conntrack GC loop as packetparser plugin is not enabled"
ts=2025-10-14T08:47:54.666Z level=info caller=dropreason/dropreason_linux.go:87 msg="Bypassing lookup IP of interest"
ts=2025-10-14T08:47:54.666Z level=info caller=manager/server.go:83 msg="starting server" name="health probe" addr=[::]:18081
ts=2025-10-14T08:47:54.666Z level=info caller=server/server.go:69 msg="starting HTTP server... on " addr=0.0.0.0:10093
ts=2025-10-14T08:47:55.964Z level=info caller=dropreason/dropreason_linux.go:125 msg="DropReason metric compiled"
ts=2025-10-14T08:47:55.965Z level=info caller=dropreason/ebpfsetup_linux.go:66 msg="Distro check:" isMariner=false
ts=2025-10-14T08:47:55.965Z level=info caller=dropreason/ebpfsetup_linux.go:75 msg="Detected kernel" version=6.12.47
ts=2025-10-14T08:47:56.775Z level=error caller=dropreason/dropreason_linux.go:156 msg="Error loading eBPF programs" error="field InetCskAcceptFexit: program inet_csk_accept_fexit: load program: permission denied: func 'inet_csk_accept' doesn't have 4-th argument: invalid bpf_context access off=24 size=8 (5 line(s) omitted)"
ts=2025-10-14T08:47:56.775Z level=info caller=server/server.go:79 msg="gracefully shutting down HTTP server..."
ts=2025-10-14T08:47:56.775Z level=info caller=server/server.go:71 msg="HTTP server stopped with err: http: Server closed"
ts=2025-10-14T08:47:56.775Z level=panic caller=controllermanager/controllermanager.go:112 msg="Error running controller manager" error="failed to reconcile plugin dropreason: failed to init plugin: field InetCskAcceptFexit: program inet_csk_accept_fexit: load program: permission denied: func 'inet_csk_accept' doesn't have 4-th argument: invalid bpf_context access off=24 size=8 (5 line(s) omitted)" errorVerbose="field InetCskAcceptFexit: program inet_csk_accept_fexit: load program: permission denied: func 'inet_csk_accept' doesn't have 4-th argument: invalid bpf_context access off=24 size=8 (5 line(s) omitted)\nfailed to init plugin\ngithub.com/microsoft/retina/pkg/managers/pluginmanager.(*PluginManager).Reconcile\n\t/go/src/github.com/microsoft/retina/pkg/managers/pluginmanager/pluginmanager.go:107\ngithub.com/microsoft/retina/pkg/managers/pluginmanager.(*PluginManager).Start\n\t/go/src/github.com/microsoft/retina/pkg/managers/pluginmanager/pluginmanager.go:159\ngithub.com/microsoft/retina/pkg/managers/controllermanager.(*Controller).Start.func1\n\t/go/src/github.com/microsoft/retina/pkg/managers/controllermanager/controllermanager.go:102\ngolang.org/x/sync/errgroup.(*Group).Go.func1\n\t/go/pkg/mod/golang.org/x/sync@v0.16.0/errgroup/errgroup.go:93\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1700\nfailed to reconcile plugin dropreason\ngithub.com/microsoft/retina/pkg/managers/pluginmanager.(*PluginManager).Start\n\t/go/src/github.com/microsoft/retina/pkg/managers/pluginmanager/pluginmanager.go:163\ngithub.com/microsoft/retina/pkg/managers/controllermanager.(*Controller).Start.func1\n\t/go/src/github.com/microsoft/retina/pkg/managers/controllermanager/controllermanager.go:102\ngolang.org/x/sync/errgroup.(*Group).Go.func1\n\t/go/pkg/mod/golang.org/x/sync@v0.16.0/errgroup/errgroup.go:93\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1700"
panic: Error running controller manager [recovered]
        panic: Error running controller manager

goroutine 46 [running]:
github.com/microsoft/retina/pkg/telemetry.TrackPanic()
        /go/src/github.com/microsoft/retina/pkg/telemetry/telemetry.go:149 +0x21b
panic({0x5f0e940?, 0xc0027e7340?})
        /usr/local/go/src/runtime/panic.go:792 +0x132
go.uber.org/zap/zapcore.CheckWriteAction.OnWrite(0x4e?, 0xc000e91d38?, {0x56cf060?, 0xc000807d60?, 0xc000e05cd8?})
        /go/pkg/mod/go.uber.org/zap@v1.27.0/zapcore/entry.go:196 +0x54
go.uber.org/zap/zapcore.(*CheckedEntry).Write(0xc000a2c750, {0xc00286e180, 0x1, 0x1})
        /go/pkg/mod/go.uber.org/zap@v1.27.0/zapcore/entry.go:262 +0x24a
go.uber.org/zap.(*Logger).Panic(0xc000e84c80?, {0x6b2ea1d?, 0x0?}, {0xc00286e180, 0x1, 0x1})
        /go/pkg/mod/go.uber.org/zap@v1.27.0/logger.go:285 +0x4b
github.com/microsoft/retina/pkg/managers/controllermanager.(*Controller).Start(0xc0009dae10, {0x7105dd0?, 0xc0009e7ef0?})
        /go/src/github.com/microsoft/retina/pkg/managers/controllermanager/controllermanager.go:112 +0x2bc
created by github.com/microsoft/retina/cmd/standard.(*Daemon).Start in goroutine 1
        /go/src/github.com/microsoft/retina/cmd/standard/daemon.go:313 +0x2caf
```

**To Reproduce**
Deploy retina to K8s cluster running nodes with Flatcar Alpha (or likely any OS with 6.12 kernel).

Seems deploying on a cluster creating with `minikube` where host is running newer enough kernel is sufficient to trigger the panic.

**Expected behavior**
Agent to start and be running.

**Platform (please complete the following information):**

- OS: Flatcar Linux Alpha
- Kubernetes Version: v1.32.7
- Retina Version: v1.0.0-rc3

** More details **
This seems to be related to the following change in kernel https://github.com/torvalds/linux/commit/92ef0fd55ac80dfc2e4654edfe5d1ddfa6e070fe (thanks @jepio).

--- a/pkg/plugin/dropreason/_cprog/drop_reason.c
+++ b/pkg/plugin/dropreason/_cprog/drop_reason.c
@@ -12,6 +12,19 @@
 #include "dynamic.h"
 #include "retina_filter.c"
 
+// CO-RE flavor struct for Linux 6.10+ where inet_csk_accept changed signature from
+// (struct sock *sk, int flags, int *err, bool kern) to
+// (struct sock *sk, struct proto_accept_arg *arg).
+// The ___new suffix is stripped during BTF type lookup, so bpf_core_type_exists()
+// checks for 'proto_accept_arg' in the running kernel's BTF.
+// https://github.com/torvalds/linux/commit/92ef0fd55ac80dfc2e4654edfe5d1ddfa6e070fe
+struct proto_accept_arg___new {
+    int flags;
+    int err;
+    int is_empty;
+    bool kern;
+} __attribute__((preserve_access_index));
+
 char __license[] SEC("license") = "Dual MIT/GPL";
 
 #define ETH_P_IP 0x0800
@@ -358,7 +371,7 @@ int BPF_PROG(tcp_v4_connect_fexit, struct sock *sk, struct sockaddr *uaddr, int 
 }
 
 SEC("kprobe/inet_csk_accept")
-int BPF_KPROBE(inet_csk_accept, struct sock *sk, int flags, int *err, bool kern)
+int BPF_KPROBE(inet_csk_accept)
 {
     /*
     This function will save the reference value to error.
@@ -366,7 +379,20 @@ int BPF_KPROBE(inet_csk_accept, struct sock *sk, int flags, int *err, bool kern)
     */
     __u64 pid_tgid = bpf_get_current_pid_tgid();
     __u32 pid = pid_tgid >> 32;
-    __u64 err_ptr = (__u64)err;
+
+    __u64 err_ptr = 0;
+
+    // Linux 6.10+: inet_csk_accept(struct sock *sk, struct proto_accept_arg *arg)
+    // err is an inline int field inside proto_accept_arg.
+    if (bpf_core_type_exists(struct proto_accept_arg___new)) {
+        struct proto_accept_arg___new *arg = (struct proto_accept_arg___new *)PT_REGS_PARM2(ctx);
+        err_ptr = (__u64)&arg->err;
+    } else {
+        // Pre-6.10: inet_csk_accept(struct sock *sk, int flags, int *err, bool kern)
+        int *err = (int *)PT_REGS_PARM3(ctx);
+        err_ptr = (__u64)err;
+    }
+
     bpf_map_update_elem(&retina_dropreason_accept_pids, &pid, &err_ptr, BPF_ANY);
     return 0;
 }
@@ -415,16 +441,22 @@ int BPF_KRETPROBE(inet_csk_accept_ret, struct sock *sk)
 }
 
 SEC("fexit/inet_csk_accept")
-int BPF_PROG(inet_csk_accept_fexit, struct sock *sk, int flags, int *err, struct sock *retsk)
+int BPF_PROG(inet_csk_accept_fexit)
 {
-    if (retsk != NULL) {
-        return 0;
+    // Each branch must be fully self-contained so the compiler generates a real
+    // branch instruction. If merged into a select, the verifier rejects the
+    // variable-offset ctx access ("dereference of modified ctx ptr").
+    //
+    // fexit ctx layout: [param1, param2, ..., paramN, return_value]
+    // Linux 6.10+: 2 params (sk, arg) -> retsk at ctx[2]
+    // Pre-6.10:    4 params (sk, flags, err, kern) -> retsk at ctx[4]
+    if (bpf_core_type_exists(struct proto_accept_arg___new)) {
+        if ((struct sock *)ctx[2] == NULL)
+            update_metrics_map_basic(TCP_ACCEPT_BASIC, 0, 0);
+    } else {
+        if ((struct sock *)ctx[4] == NULL)
+            update_metrics_map_basic(TCP_ACCEPT_BASIC, 0, 0);
     }
-
-    // TODO
-    // Pass 0 packet length - get_packet_from_sock above doesn't obtain this value, either.
-    // Pass 0 return value; verifier failure, same as buggy kprobe above.
-    update_metrics_map_basic(TCP_ACCEPT_BASIC, 0, 0); 
 
     return 0;
 }

